integration/src/test/kotlin/kotlinx/benchmark/integration/SuiteSourceGeneratorTest.kt

```diff
@@ -20,4 +21,38 @@ class SuiteSourceGeneratorTest : GradleTest() {
             }
         }
     }
+
+    @Test
+    fun generateAndCompileNativeBenchmarks() {
+        generateAndCompile("native")
+    }
+
+    @Test
+    fun generateAndCompileJSBenchmarks() {
+        generateAndCompile("jsIr")
+    }
+
+    @Test
+    fun generateAndCompileWasmBenchmarks() {
+        generateAndCompile("wasmJs")
+    }
+
+    private fun generateAndCompile(target: String) {
+        project("kotlin-multiplatform", true).let { runner ->
+            runner.run(":${target}BenchmarkGenerate")
+
+            runner.generatedDir(target, "RootBenchmark_Descriptor.kt") { descriptorFile ->
+                val text = descriptorFile.readText()
+                assertFalse(
+                    text.contains("<root>"),
+                    "Generated descriptor contains illegal characters '<root>' in $text"
+                )
+            }
+
+            val capitalizedTarget = target.replaceFirstChar {
+                if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()
+            }
```

Comment:
```suggestion
            val capitalizedTarget = target.replaceFirstChar { it.uppercaseChar() }
```

`import java.util.*` would also be redundant.